bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBluetoothDevice.java

```diff
@@ -0,0 +1,474 @@
+        if (c == null) {
+            logger.warn("Characteristic '{}' is missing on device '{}'.", characteristic.getUuid(), address);
+            return false;
+        }
+
+        scheduler.submit(() -> {
+            try {
+                c.writeValue(characteristic.getByteValue(), null);
+                notifyListeners(BluetoothEventType.CHARACTERISTIC_WRITE_COMPLETE, characteristic,
+                        BluetoothCompletionStatus.SUCCESS);
+
+            } catch (DBusException e) {
+                logger.debug("Exception occurred when trying to write characteristic '{}': {}",
+                        characteristic.getUuid(), e.getMessage());
+                notifyListeners(BluetoothEventType.CHARACTERISTIC_WRITE_COMPLETE, characteristic,
+                        BluetoothCompletionStatus.ERROR);
+            }
+        });
+        return true;
+    }
+
+    @Override
+    public void onDBusBlueZEvent(BlueZEvent event) {
+        logger.debug("Unsupported event: {}", event);
+    }
+
+    @Override
+    public void onServicesResolved(ServicesResolvedEvent event) {
+        if (event.isResolved()) {
+            notifyListeners(BluetoothEventType.SERVICES_DISCOVERED);
+        }
+    }
+
+    @Override
+    public void onNameUpdate(NameEvent event) {
+        BluetoothScanNotification notification = new BluetoothScanNotification();
+        notification.setDeviceName(event.getName());
+        notifyListeners(BluetoothEventType.SCAN_RECORD, notification);
+    }
+
+    @Override
+    public void onManufacturerDataUpdate(ManufacturerDataEvent event) {
+        for (Map.Entry<Short, byte[]> entry : event.getData().entrySet()) {
+            BluetoothScanNotification notification = new BluetoothScanNotification();
+            byte[] data = new byte[entry.getValue().length + 2];
+            data[0] = (byte) (entry.getKey() & 0xFF);
+            data[1] = (byte) (entry.getKey() >>> 8);
+
+            System.arraycopy(entry.getValue(), 0, data, 2, entry.getValue().length);
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("Received manufacturer data for '{}': {}", address, HexUtils.bytesToHex(data, " "));
+            }
+
+            notification.setManufacturerData(data);
+            notifyListeners(BluetoothEventType.SCAN_RECORD, notification);
+        }
+    }
+
+    @Override
+    public void onTxPowerUpdate(TXPowerEvent event) {
+        this.txPower = (int) event.getTxPower();
+    }
+
+    @Override
+    public void onCharacteristicNotify(CharacteristicUpdateEvent event) {
+        // Here it is a bit special - as the event is linked to the DBUS path, not characteristic UUID.
+        // So we need to find the characteristic by its DBUS path.
+        BluetoothGattCharacteristic characteristic = getDBusBlueZCharacteristicByDBusPath(event.getDbusPath());
+        if (characteristic == null) {
+            logger.info("Received a notification for a characteristic not found on device.");
```

Comment:
What is this info telling to the user? Either he can ignore it, then debug should suffice or it is a problem in the configuration/device, then it should rather be a warning with more details.

Reply:
The situation is unusual, and might indicate a problem. But the issue doesn't require intervention so I'll change to debug.